### PR TITLE
fix: add missing @opentelemetry/core dependency

### PR DESCRIPTION
**Pull request by pichlermarc**
## Which problem is this PR solving?

See #2472, we're using `@opentelemetry/core` but don't have it listed as a dependency in some packages.

## Short description of the changes

- adds the dependency to the packages where it's missing (identified through comparing https://github.com/open-telemetry/opentelemetry-js-contrib/compare/auto-instrumentations-node-v0.50.0...auto-instrumentations-node-v0.50.1)
- pins the `@opentelemetry/semantic-conventions` dependency for `@opentelemetry/instrumentation-pg`  as it's using the `incubating` entrypoint, which may break on minor bumps.

### CODE REVIEW

**Review by maryliag — APPROVED** [review]
Thanks for fixing this!

File: plugins/node/opentelemetry-instrumentation-pg/package.json [hunk 1]
```diff
@@ -73,8 +73,9 @@
   },
   "dependencies": {
     "@opentelemetry/instrumentation": "^0.53.0",
-    "@opentelemetry/semantic-conventions": "^1.27.0",
+    "@opentelemetry/semantic-conventions": "1.27.0",
```
Comment (pichlermarc) [inline plugins/node/opentelemetry-instrumentation-pg/package.json:76]: drive-by-fix: we're using the `@opentelemetry/semantic-conventions/incubating` entrypoint which may break in minor versions - pinning this to avoid any surprises when we release the `1.28.0`.

Comment (trentm) [reply plugins/node/opentelemetry-instrumentation-pg/package.json:76]: Might want the same at https://github.com/open-telemetry/opentelemetry-js-contrib/blob/645ac2e9fa322503e88283985f3f3bb2349a9729/plugins/node/opentelemetry-instrumentation-aws-lambda/package.json#L63

We should have a lint check for this perhaps.

**Review by trentm — APPROVED** [review]
